Comestible Nutrient Rebalance (22 of 22)

KorGgenT opened the pull request:
<!--
### How to use
Leave the headings unless they don't apply to your PR, replace commented out text ( surrounded with <!-- and -​-> ) with text describing your PR.
-->

#### Summary
<!--
A one-line description of your change that will be extracted and added to the project changelog at https://github.com/CleverRaven/Cataclysm-DDA/blob/master/data/changelog.txt
The format is: ```SUMMARY: Category "description"```  
The categories to choose from are: Features, Content, Interface, Mods, Balance, Bugfixes, Performance, Infrastructure, Build, I18N  
Example: ```SUMMARY: Content "Adds new mutation category 'Mouse'"```
See the Changelog Guidelines at https://github.com/CleverRaven/Cataclysm-DDA/blob/master/doc/CHANGELOG_GUIDELINES.md for explanations of the categories.
-->
```SUMMARY: Balance "rebalance comestibles' calories, vitamins, weight, and volume"```
#### Purpose of change
<!--
If there's an existing issue describing the problem this PR addresses or the feature it adds, please link it like: ```#1234```  
If it *fully* resolves an issue, link it like: ```Fixes #1234```  
Even if the issue describes the problem, please provide a few-sentence summary here.  
Example: ```Fixes #1234 - XL mutants cannot wear arm/leg splints due to missing OVERSIZE flag.```  
If there is no related issue, please describe the issue you are addressing, including how to trigger a bug if this is a bugfix.
-->
It has come to my attention that a very large portion of comestibles do not have correct nutrient values -- both calories and vitamins. I have decided to undertake (with hopefully some help) going over all comestibles' nutrient values and matching them to real-life nutrition facts (marloss berries notwithstanding!)

This is where I will keep all of my progress together -- so I am able to see which files have been gone over, and which still have yet to be gone over. Once these are all done, I'll make all the actual pull requests -- but i will probably hold off (at least mostly) until i take a look at recipes, too, since this will affect them in a major way.

This overhaul will be in two large parts - comestible changes and recipe changes.

This will cover both calories and vitamins. 
Items may warrant a portion size change, and/or weight and volume change.
[This is a link to a working spreadsheet with the json files in them and links to source material](https://docs.google.com/spreadsheets/d/1n1roPzIsRvABg3Ry0q7SPxnUK_C0GR9cjK27m4cF9ok/edit#gid=475503977)

Fixes #26050

- [x] alcohol
- [x] bread
- [X] brewing (no nutrition to speak of)
- [x] carnivore
- [x] dairy
- [x] drink
- [x] drink_other
- [x] egg
- [x] frozen
- [x] fruit_dishes
- [X] irradiated_fruit (utilizes copy-from)
- [X] irradiated_veggy (utilizes copy-from)
- [x] junkfood
- [x] meat_dishes
- [X] meat_dishes_human (utilizes copy-from)
- [X] med (this is not food, so no nutrition to speak of)
- [X] mre (very, very generic)
- [X] mushroom
- [X] mutagen (this is not food, so no nutrition to speak of)
- [x] nuts
- [x] offal_dishes (i originally added this one myself)
- [x] other
- [x] protein
- [x] raw_fruit
- [x] raw_veggy
- [x] sandwich
- [x] seed
- [x] soup
- [x] spice
- [x] veggy_dishes
- [x] wheat


NOTE: This PR is related to #27542

NOTE2: This PR is the last in a group of PRs. it contains some smaller file changes that were too small to make a whole PR for.

anothersimulacrum commented on data/json/items/comestibles/alcohol.json:
```diff
@@ -1060,19 +1062,20 @@
     "price": 1000,
     "material": [ "junk", "honey" ],
     "volume": 1,
+    "charges": 2,
```
Because you changed the charges of this item, you need to divide the weight and price by the number of charges.

KorGgenT replied:
i don't know about that, for this item. it still gets you just as drunk.

anothersimulacrum commented on data/json/items/comestibles/bread.json:
```diff
@@ -86,100 +86,102 @@
     "price": 140,
     "material": "wheat",
-    "volume": 2,
-    "charges": 6,
```
Because you changed the charges of this item, you need to divide the weight and price by the number of charges.

In bread.json, you change charges on lines (other than this):
119, 141, 162, 184, 204, 226, and 244

anothersimulacrum commented on data/json/items/comestibles/dairy.json:
```diff
@@ -63,41 +65,43 @@
     "material": [ "milk" ],
     "volume": 1,
-    "charges": 3,
+    "charges": 7,
```
Because you changed the charges of this item, you need to divide the weight and price by the number of charges.

In alcohol.json, you change charges on lines (other than this one):
102

anothersimulacrum commented on data/json/items/comestibles/drink.json:
```diff
@@ -210,26 +197,28 @@
     "material": [ "milk", "junk" ],
     "volume": 1,
     "phase": "liquid",
+    "charges": 10,
```
Because you changed the charges of this item, you need to divide the weight and price by the number of charges.

anothersimulacrum commented on data/json/items/comestibles/mushroom.json:
```diff
@@ -99,77 +81,49 @@
-    "flags": [ "EDIBLE_FROZEN" ],
-    "charges": 4,
-    "fun": 1
+    "charges": 4
```
Because you changed the charges of this item, you need to divide the weight and price by the number of charges.

anothersimulacrum commented on data/json/items/comestibles/nuts.json:
```diff
@@ -4,62 +4,56 @@
     "volume": 1,
     "flags": [ "EDIBLE_FROZEN" ],
-    "charges": 2,
+    "charges": 4,
```
Because you changed the charges of this item, you need to divide the weight and price by the number of charges.

In nuts.json, you change charges on lines (other than this one):
44, 80, 114, 137, 171, 194, 214, 247, 294, 328, 385, and 416

KorGgenT replied:
not really sure a price change is necessary here, as the weights didn't really change much. only the volume changed here.

anothersimulacrum commented on data/json/items/comestibles/soup.json:
```diff
@@ -51,89 +51,89 @@
     "material": "veggy",
     "volume": 1,
     "phase": "liquid",
-    "charges": 4,
```
Because you changed the charges of this item, you need to divide the weight and price by the number of charges.

In soup.json, you change charges on lines (other than this one):
68, 90, 112, 135, 180, 197, 210, 278, and 298

KorGgenT replied:
for these the serving size didn't change, only the volume.

anothersimulacrum commented on data/json/items/comestibles/wheat.json:
```diff
@@ -12,239 +12,247 @@
-    "volume": 4,
+    "volume": 1,
     "flags": [ "EDIBLE_FROZEN" ],
-    "charges": 4,
```
Because you changed the charges of this item, you need to divide the weight and price by the number of charges.

In wheat.json, you change charges on lines (other than this one):
41, 98, 138, 200, 319, 339, 394, 441, 463, 487

anothersimulacrum commented:
I figured you had already adjusted the calories to be accurate per charge (given that this is the nutrient rebalance), and in the cases where you changed the weight, the weight. You just hadn't modified the price for any or the weight for some, so I just wanted to bring your attention to that.

KorGgenT commented:
@anothersimulacrum I have made a few changes, and justified a few other non-changes. I hope they are to your liking. Thank you for keeping me honest

KorGgenT commented:
Now that I have fruits and veggies finished, once this PR is merged #26799 will no longer have a use

KorGgenT commented:
My first pass through the entirety of the items is complete. I have updated the calories and vitamins for all of them. I have updated a few prices, but I have not gone over them with a fine-toothed comb.
